py3status/modules/mpris.py

```diff
@@ -0,0 +1,422 @@
+    icon_stop = '◾'
+    icon_next = '»'
+    icon_previous = '«'
+    show_controls = None
+    state_pause = '▮▮'
+    state_play = '▶'
+    player_priority = None
+
+    def _get_player(self, player):
+        """
+        Get dbus object
+        """
+        try:
+            return self._dbus.get(SERVICE_BUS + '.%s' % player, SERVICE_BUS_URL)
+        except Exception:
+            return None
+
+    def _get_state(self, player):
+        """
+        Get the state of a player
+        """
+        player = self._get_player(player)
+        if player is None:
+            return 'None'
+
+        return player.PlaybackStatus
+
+    def _get_state_format(self, i3s_config):
+        playback_status = self._player.PlaybackStatus
+
+        if playback_status == 'Playing':
+            color = self.color_playing or i3s_config['color_good']
+            state = self.state_play
+        elif playback_status == 'Paused':
+            color = self.color_paused or i3s_config['color_degraded']
+            state = self.state_pause
+        else:
+            color = self.color_stopped or i3s_config['color_bad']
+            state = self.state_stop
+
+        return (color, state)
+
+    def _detect_running_player(self):
+        """
+        Detect running player process, if any
+        """
+        players_paused = []
+        players_stopped = []
+        players = []
+        players_prioritized = []
+
+        _dbus = self._dbus.get('org.freedesktop.DBus')
+        for player in _dbus.ListNames():
+            if SERVICE_BUS in player:
+                player = re.sub(r'%s' % SERVICE_BUS_REGEX, '', player)
+                players.append(player)
+
+        if self.player_priority is None:
+            players_prioritized = players
+        else:
+            player_priority = self.player_priority.split(',')
+
+            for player_name in player_priority:
+                if player_name in players:
+                    players.remove(player_name)
+                    players_prioritized.append(player_name)
+            if '*' in player_priority:
+                players_prioritized += players
+
+        for player_name in players_prioritized:
+            player_state = self._get_state(player_name)
+
+            if player_state == 'Playing':
+                return player_name
+            elif player_state == 'Paused':
+                players_paused.append(player_name)
+            else:
+                players_stopped.append(player_name)
+
+        if players_paused:
+            return players_paused[0]
+        if players_stopped:
+            return players_stopped[0]
+
+        return 'None'
+
+    def _get_text(self, i3s_config):
+        """
+        Get the current metadatas
+        """
+        is_stream = False
+        player = UNKNOWN
+        state = UNKNOWN
+        artist = UNKNOWN
+        album = UNKNOWN
+        length = UNKNOWN_TIME
+        ptime = UNKNOWN_TIME
+        title = UNKNOWN
+
+        _left_s = None
```

Comment:
@ritze generally in python a leading `_` has special meaning.  It shows that the thing should be treated as private.  Python has no real private as we can just find anything in a module/class etc, but the underscore let's people know that they should not be playing with stuff, or that it is not supported.

In Py3status objects we use it for methods names to stop them being run to get a modules output too.

`_left_s` is a local variable so it is effectively private anyhow so there is no need to have the leading underscore.  If you want to keep it that's fine but reading the code it just looks a bit out of place.